src/Highway.Data.EntityFramework/Factories/DomainRepositoryFactory.cs

```diff
@@ -9,7 +9,7 @@ namespace Highway.Data.Factories
     /// <summary>
     /// Simple factory for constructing repositories
     /// </summary>
-    public class DomainRepositoryFactory : IDomainRepositoryFactory
+    public class DomainRepositoryFactory : IDomainRepositoryFactory, IReadonlyDomainRepositoryFactory
```

Comment:
Do you prefer to have additional methods in a single DomainRepositoryFactory, or have a ReadonlyDomainRepository factory?  This seems like the right place for the fact of a read-only repository to be discoverable, but I can go either way.

Reply:
I like the discoverability of it here.